docs/implementation.md

```diff
@@ -0,0 +1,210 @@
+## Implementing Links
+
+Before implementing your own Link, be sure to read the [usage](usage.md) document.
+
+Links implement a `request` method that takes an operation and returns the results in an Observable.
+The apollo-link's `Observable` follows the ECMAScript [proposal](https://github.com/tc39/proposal-observable#api) with some extensions provided by [zen-observable](https://github.com/zenparsing/zen-observable#api).
```

Comment:
The first thought was to keep `asPromiseWrapper` out of apollo-link, since simple queries can be done with [apollo-fetch](https://github.com/apollographql/apollo-fetch). You make a good point that it's worth adding `asPromiseWrapper` back in. Any reasons not to @jbaxleyiii